src/outputs/outputs.cpp

```diff
@@ -235,6 +224,26 @@ Outputs::Outputs(Mesh *pm, ParameterInput *pin, SimTime *tm) {
       // read cartesian mapping option
       op.cartesian_vector = false;
 
+      // read single precision output option
+      bool is_hdf5_output = (op.file_type.compare("rst") == 0) ||
+                            (op.file_type.compare("ath5") == 0) ||
+                            (op.file_type.compare("hdf5") == 0);
```

Comment:
I would write this differently,

```
std::unordered_set<std::string> hdf5_extensions = {"rst", "ath5", "hdf5"};
const bool is_hdf5_output = hdf5_extensions.count(op.file_type) == 1;
```
Looks like in c++20 set gets a contains keyword that would be nicer but guess we aren't there.

Reply:
I don't think using a `set` or `undordered_set` makes sense here. That's too much overhead to check 3 possibilities. If the number of options explodes, we can revisit this. But I am removing the `.compare`. Not sure why that was used...